designdocs/recovery_system.md

```diff
@@ -0,0 +1,476 @@
+
+## System Workflow
+
+### 1. Account Creation & Recovery Secret Issuance
+
+When a user first creates an account (MSA) through a wallet provider, the provider will ask the blockchain to generate and register a Recovery Secret for that user as part of the setup. The sequence diagram below illustrates this process:
+
+```mermaid
+sequenceDiagram
+    participant U as User
+    participant App as Provider App
+    participant EW as Provider Wallet
+    participant SDK as Frequency SDK
+    participant B as App Backend
+    participant G as Gateway API
+    participant F as Frequency
+
+    Note over U,F: Account Creation
+    U->>App: Enter Email
+    App->>EW: Initiate Provider Wallet Account Creation
+    EW->>U: Send OTP email
+    U->>App: Enter OTP
+    App->>EW: Submit OTP
+    EW->>App: Confirm Provider Wallet ID and address
+    App->>U: Request handle and delegations
+    U->>App: Enter handle and confirm delegations
+    App->>EW: Create second wallet for Frequency Control Key (secp256k1)
+    EW->>App: Return Control Key Public Key
+    App->>SDK: create_msa_for_email(controlKey, email)
+    SDK->>B: Request payloads for account creation
+    B->>G: Request payloads for account creation
+    G->>B: Return addProviderPayload, claimHandlePayload, graph key setup payload, recoveryTokenPayload
+    B->>SDK: Return addProviderPayload, claimHandlePayload, graph key setup payload, recoveryTokenPayload
+    Note over App,SDK: Signature request may need to be mediated by App
+    SDK->>EW: Request signatures for payloads
+    EW->>SDK: Return signatures
+    SDK->>B: Submit signed payloads
+    B->>G: *NEW* Create MSA with recovery email
+    G->>B: Return generated graph key and recovery code
+    G->>F: Submit batch transaction
+    B->>+G: (Polling) Get MSA
+    G->>F: Check for MSA
+    G->>-B: MSA Confirmation
+    B->>B: Store MSA, Graph Key, Recovery Code
+    B->>G: Get full handle
+    G->>F: handleForMSA
+    F->>G: Return handle
+    G->>B: Return handle
+    B->>App: Return handle
+    App->>U: Show confirmed handle
+    B->>U: (Async) Send Frequency intro/recovery email including recovery code
+```
+
+#### Sequence diagram – Account creation and Recovery Secret setup
+
+1. User Account Creation: The user signs up with a wallet/app (the provider) and provides their Authentication Contact as part of registration. The provider creates a new MSA on the Frequency blockchain for the user. During this on-chain account creation, the user’s initial Control Key (public key) is linked to the MSA (this may be a key managed by the provider).
+
+2. Recovery Secret Generation: The blockchain generates a random secret (e.g. a 256-bit entropy value). It will call a secure random generator to get 32 bytes and then encode as a hexadecimal phrase.
```

Comment:
blockchain can not generate a secure random secret as far as I know. This has to be generated offchain